src/include/miopen/tensorOp/problem_description.hpp

```diff
@@ -0,0 +1,132 @@
+                       const void* beta_,
+                       const TensorDescriptor& aTensorDesc_,
+                       const TensorDescriptor& bTensorDesc_,
+                       const TensorDescriptor& cTensorDesc_,
+                       const bool nonStandardSquash_)
+        : tensorOp(tensorOp_),
+          aTensorDesc(aTensorDesc_),
+          bTensorDesc(bTensorDesc_),
+          cTensorDesc(cTensorDesc_),
+          nonStandardSquash(nonStandardSquash_)
+    {
+        if(beta_ == nullptr)
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "Beta value is nullptr");
+        }
+
+        beta = *(static_cast<const float*>(beta_));
+
+        if(aTensorDesc.GetElementSize() != cTensorDesc.GetElementSize())
+        {
+            MIOPEN_THROW("A and C Tensors do not match");
+        }
+
+        if(bTensorDesc.GetType() != cTensorDesc.GetType())
+        {
+            MIOPEN_THROW("Datatypes for B and C tensors do not match !");
+        }
+
+        auto blens = bTensorDesc.GetLengths();
+        auto clens = cTensorDesc.GetLengths();
+
+        if(clens.size() > 5)
+        {
+            MIOPEN_THROW("Tensor dimension larger than 5: " + std::to_string(clens.size()));
+        }
+
+        if(blens.size() != clens.size())
+        {
+            MIOPEN_THROW("Number of dims in B and C Tensors do not match: " +
+                         std::to_string(blens.size()) + ", " + std::to_string(clens.size()));
+        }
+
+        if(!nonStandardSquash)
+        {
+            for(std::size_t i = 0; i < clens.size(); i++)
+            {
+                if(blens[i] != 1 && blens[i] != clens[i])
+                {
+                    MIOPEN_THROW("BTensor dim != 1 && BTensor dim != CTensor dim: " +
+                                 std::to_string(i));
+                }
+            }
+        }
+        else
+        {
+            // non standard behavior because blens[1] can be not equalt to clens[1]
+            if(!(clens.size() == 3 && blens[0] == 1 && clens[0] == 1 && blens[2] == clens[2]))
+            {
+                MIOPEN_THROW(
+                    "Non standard squashed operation supported only for 3d tensors and for "
+                    "the specific configuration");
+            }
+        }
+    }
+
+    miopenTensorOp_t GetTensorOp() const { return tensorOp; }
+
+    float GetBeta() const { return beta; }
+
+    const TensorDescriptor& GetATensorDesc() const { return aTensorDesc; }
+    const TensorDescriptor& GetBTensorDesc() const { return bTensorDesc; }
+    const TensorDescriptor& GetCTensorDesc() const { return cTensorDesc; }
+
+    bool GetNonStandardSquash() const { return nonStandardSquash; }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+private:
+    const miopenTensorOp_t tensorOp;
+
+    float beta;
+
+    const TensorDescriptor& aTensorDesc;
+    const TensorDescriptor& bTensorDesc;
+    const TensorDescriptor& cTensorDesc;
```

Comment:
We should keep tensor descriptors alive
```suggestion
    TensorDescriptor aTensorDesc;
    TensorDescriptor bTensorDesc;
    TensorDescriptor cTensorDesc;
```